build/debs/build.go

```diff
@@ -241,177 +234,214 @@ func fetchVersion(url string) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
 	// Remove a newline and the v prefix from the string
 	return strings.Replace(strings.Replace(string(versionBytes), "v", "", 1), "\n", "", 1), nil
 }
 
+func getPackageVersion(packageDef packageDefinition) (string, error) {
+	log.Printf("package name is %s", packageDef.Name)
+	switch packageDef.Name {
+	case "kubernetes-cni":
+		return getCNIVersion(packageDef)
+	case "cri-tools":
+		return getCRIToolsVersion(packageDef)
+	}
+
+	log.Printf("using Kubernetes version")
+	return packageDef.KubernetesVersion, nil
+}
+
+func getKubernetesVersion(packageDef packageDefinition) (string, error) {
+	if packageDef.KubernetesVersion != "" {
+		return packageDef.KubernetesVersion, nil
+	}
+	switch packageDef.Distribution {
+	case DistributionTesting:
+		return getTestingKubeVersion()
+	case DistributionUnstable:
+		return getUnstableKubeVersion()
+	}
```

Comment:
Style nit:  this should have the getStableKubeVersion() happen for an explicit case, plus a default case that returns an error explicitly for unknown cases.  We may over time add and remove distribution types, modifying such switch statements, so having them explicit and throw errors for unhandled cases likely reduces future bugs.

Reply:
Going further through review I'm seeing a lot of differences across the per-channel switch statements.  I suggest consistently doing explicitly:

 * case channel foo
 * case channel bar
 * case channel baz
 * default: error

for ease of maintenance.